examples/computer_vision/mnist_pl/mnist.py

```diff
@@ -1,76 +1,82 @@
-        self.dims = (1, 28, 28)
+        # Set our init args as class attributes
+        self.hidden_size = hidden_size
+        self.learning_rate = learning_rate
 
+        # Hardcode some dataset specific attributes
+        self.num_classes = 10
+        self.dims = (1, 28, 28)
+        channels, width, height = self.dims
+        self.transform = transforms.Compose([
+            transforms.ToTensor(),
+            transforms.Normalize((0.1307,), (0.3081,))
+        ])
+
+        # Define PyTorch model
+        self.model = nn.Sequential(
+            nn.Flatten(),
+            nn.Linear(channels * width * height, hidden_size),
+            nn.ReLU(),
+            nn.Dropout(0.1),
+            nn.Linear(hidden_size, hidden_size),
+            nn.ReLU(),
+            nn.Dropout(0.1),
+            nn.Linear(hidden_size, self.num_classes)
+        )
 
     def forward(self, x):
-        batch_size, channels, width, height = x.size()
-
-        # (b, 1, 28, 28) -> (b, 1*28*28)
-        x = x.view(batch_size, -1)
-
-        # layer 1 (b, 1*28*28) -> (b, 128)
-        x = self.layer_1(x)
-        x = torch.relu(x)
-
-        # layer 2 (b, 128) -> (b, 256)
-        x = self.layer_2(x)
-        x = torch.relu(x)
-
-        # layer 3 (b, 256) -> (b, 10)
-        x = self.layer_3(x)
+        x = self.model(x)
+        return F.log_softmax(x, dim=1)
 
-        # probability distribution over labels
-        x = torch.log_softmax(x, dim=1)
-
-        return x
-
-    def _loss_fn(self, logits, labels):
-        return F.nll_loss(logits, labels)
-
-    def training_step(self, batch, *args, **kwargs):
+    def training_step(self, batch, batch_idx):
         x, y = batch
-        logits = self.forward(x)
-        loss = self._loss_fn(logits, y)
+        logits = self(x)
+        loss = F.nll_loss(logits, y)
         self.log('train_loss', loss)
         return {'loss': loss}
 
-    def validation_step(self, batch, batch_idx, *args, **kwargs):
+    def validation_step(self, batch, batch_idx):
         x, y = batch
-        logits = self.forward(x)
-        loss = self._loss_fn(logits, y)
-        self.log('val_loss', loss)
+        logits = self(x)
+        loss = F.nll_loss(logits, y)
+        preds = torch.argmax(logits, dim=1)
+        acc = accuracy(preds, y)
+
+        return {'val_loss': loss, 'accuracy': acc}
 
-        pred = logits.argmax(dim=1, keepdim=True)
-        accuracy = pred.eq(y.view_as(pred)).sum().item() / len(x)
-        return {'val_loss': loss, 'accuracy': accuracy}
+    def test_step(self, batch, batch_idx):
+        # Here we just reuse the validation_step for testing
+        return self.validation_step(batch, batch_idx)
 
     def configure_optimizers(self):
-        optimizer = torch.optim.Adam(self.parameters(),
-                                     lr=self.hparams.lr)
+        optimizer = torch.optim.Adam(self.parameters(), lr=self.learning_rate)
         return optimizer
-
```

Comment:
could we keep this feature? to show that it is a standalone fully functioning pytorch_lightning experiment?

Reply:
done